clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -2857,32 +2857,59 @@ static void handleWeakImportAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
 }
 
 // Checks correctness of mutual usage of different work_group_size attributes:
-// reqd_work_group_size, max_work_group_size. Values of reqd_work_group_size
-// arguments shall be equal or less than values coming from max_work_group_size.
+// reqd_work_group_size, max_work_group_size and max_global_work_dim.
+// Values of reqd_work_group_size arguments shall be equal or less than values
+// coming from max_work_group_size.
+// In case of 'max_global_work_dim' attribute equals to 0 we shall ensure
+// that if max_work_group_size and reqd_work_group_size attributes exist,
+// then they are equal (1, 1, 1).
 static bool checkWorkGroupSizeValues(Sema &S, Decl *D, const ParsedAttr &Attr,
                                      uint32_t WGSize[3]) {
-  if (const SYCLIntelMaxWorkGroupSizeAttr *A =
-      D->getAttr<SYCLIntelMaxWorkGroupSizeAttr>()) {
+  bool Result = true;
+  auto checkZeroDim = [](auto &S, auto &A, auto &Attr, auto X, auto Y,
+                         auto Z) -> bool{
```

Comment:
Why all-auto?

Reply:
Why not?

Reply:
Welcome into the future! \o/

Reply:
> Why not?

Hmm.. because [don’t “almost always” use auto](https://llvm.org/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable)?

Reply:
Ok